From 5151bd3f8dfeb0aa53a7c9e7aa86ceca2f83d531 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Tue, 28 Apr 2020 12:43:07 +0100
Subject: drm/i915/selftests: Tweak the tolerance for clock ticks to 12.5%
Git-commit: 96a4faf524fcfbb6c1f2ce91238e81062895ec09
Patch-mainline: v5.8-rc1
References: jsc#SLE-12680, jsc#SLE-12880, jsc#SLE-12882, jsc#SLE-12883, jsc#SLE-13496, jsc#SLE-15322

Give a small bump for our tolerance on comparing the expected vs
measured clock ticks/time from 10% to 12.5% to accommodate a bad result
on Sandybridge that was off by 10.3%. Hopefully, that is the worst we
will see.

Closes: https://gitlab.freedesktop.org/drm/intel/-/issues/1802
Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
Reviewed-by: Mika Kuoppala <mika.kuoppala@linux.intel.com>
Link: https://patchwork.freedesktop.org/patch/msgid/20200428114307.5153-1-chris@chris-wilson.co.uk
Signed-off-by: Patrik Jakobsson <pjakobsson@suse.de>
---
 drivers/gpu/drm/i915/gt/selftest_rps.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/i915/gt/selftest_rps.c b/drivers/gpu/drm/i915/gt/selftest_rps.c
index 6ce7003bf92f..21338c1980de 100644
--- a/drivers/gpu/drm/i915/gt/selftest_rps.c
+++ b/drivers/gpu/drm/i915/gt/selftest_rps.c
@@ -312,15 +312,15 @@ int live_rps_clock_interval(void *arg)
 				engine->name, cycles, time, ktime_to_ns(dt), expected,
 				gt->clock_frequency / 1000);
 
-			if (10 * time < 9 * ktime_to_ns(dt) ||
-			    10 * time > 11 * ktime_to_ns(dt)) {
+			if (10 * time < 8 * ktime_to_ns(dt) ||
+			    8 * time > 10 * ktime_to_ns(dt)) {
 				pr_err("%s: rps clock time does not match walltime!\n",
 				       engine->name);
 				err = -EINVAL;
 			}
 
-			if (10 * expected < 9 * cycles ||
-			    10 * expected > 11 * cycles) {
+			if (10 * expected < 8 * cycles ||
+			    8 * expected > 10 * cycles) {
 				pr_err("%s: walltime does not match rps clock ticks!\n",
 				       engine->name);
 				err = -EINVAL;
-- 
2.28.0

